binlog_streamer.go

```diff
@@ -153,6 +155,9 @@ func (s *BinlogStreamer) Run() {
 		case *replication.RotateEvent:
 			// This event is needed because we need to update the last successful
 			// binlog position.
+			s.lastResumableBinlogPosition.Pos = uint32(e.Position)
+			s.lastResumableBinlogPosition.Name = string(e.NextLogName)
```

Comment:
Wait, I thought we agreed we can't resume from a RotateEvent's position?

Reply:
This is the only place where MySQL replication gives us the file name of the binlog. This value is therefore cached until the next Rotate event, which would give us the next file name. So we have to set this filename here?

Reply:
well, why can't we use a separate `currentFilename` variable to cache the current filename? 

the current filename needs a clear separation from the resumable position.